app/src/main/java/org/mozilla/rocket/component/LaunchIntentDispatcher.kt

```diff
@@ -32,19 +38,26 @@ class LaunchIntentDispatcher {
              * This extra is passed by the Notification (either [RocketMessagingService.onRemoteMessage] or System tray
              * if we have this extra, we want to show this url in a new tab
              */
-            val pushOpenUrl = intent.getStringExtra(RocketMessagingService.PUSH_OPEN_URL)
-            if (pushOpenUrl != null) {
-
-                val new = Intent(Intent.ACTION_VIEW)
-                new.data = Uri.parse(pushOpenUrl)
-                new.action = Intent.ACTION_VIEW
-                new.setClass(context, MainActivity::class.java)
-                new.putExtra(IntentUtils.EXTRA_OPEN_NEW_TAB, true)
-                context.startActivity(new)
-                return Action.HANDLED
+            intent.getStringExtra(RocketMessagingService.PUSH_OPEN_URL)?.run {
+
+                intent.data = Uri.parse(this)
+                intent.setClass(context, MainActivity::class.java)
+                intent.putExtra(IntentUtils.EXTRA_OPEN_NEW_TAB, true)
```

Comment:
`intent.action = Intent.ACTION_VIEW` is missing?

Reply:
yes!